modules/ffmpeg/producer/ffmpeg_producer.cpp

```diff
@@ -405,74 +408,77 @@ struct ffmpeg_producer : public core::frame_producer_base
 		if (is_url() || input_.loop())
 			return std::numeric_limits<uint32_t>::max();
 
-		uint32_t nb_frames = file_nb_frames();
-
-		nb_frames = std::min(length_, nb_frames - start_);
-		nb_frames = muxer_->calc_nb_frames(nb_frames);
+		auto nb_frames = std::min(input_.out(), file_nb_frames());
+		if(nb_frames >= input_.in()) nb_frames -= input_.in(); else nb_frames = 0;
 
-		return nb_frames;
+		return muxer_->calc_nb_frames(nb_frames);
 	}
 
 	uint32_t file_nb_frames() const
 	{
-		uint32_t file_nb_frames = 0;
-		file_nb_frames = std::max(file_nb_frames, video_decoder_ ? video_decoder_->nb_frames() : 0);
-		return file_nb_frames;
+		return video_decoder_ ? video_decoder_->nb_frames() : 0;
 	}
 
 	std::future<std::wstring> call(const std::vector<std::wstring>& params) override
 	{
-		static const boost::wregex loop_exp(LR"(LOOP\s*(?<VALUE>\d?)?)", boost::regex::icase);
-		static const boost::wregex seek_exp(LR"(SEEK\s+(?<VALUE>(\+|-)?\d+)(\s+(?<WHENCE>REL|END))?)", boost::regex::icase);
-		static const boost::wregex length_exp(LR"(LENGTH\s+(?<VALUE>\d+)?)", boost::regex::icase);
-		static const boost::wregex start_exp(LR"(START\s+(?<VALUE>\d+)?)", boost::regex::icase);
-
-		auto param = boost::algorithm::join(params, L" ");
-
 		std::wstring result;
 
-		boost::wsmatch what;
-		if(boost::regex_match(param, what, loop_exp))
-		{
-			auto value = what["VALUE"].str();
-			if (!value.empty())
-				input_.loop(boost::lexical_cast<bool>(value));
-			result = boost::lexical_cast<std::wstring>(input_.loop());
-		}
-		else if(boost::regex_match(param, what, seek_exp))
+		if(params.size())
 		{
-			auto value = boost::lexical_cast<int64_t>(what["VALUE"].str());
-			auto whence = what["WHENCE"].str();
-			auto total = file_nb_frames();
+			std::wstring cmd = params.at(0), value;
+			if(params.size() > 1) value = params.at(1);
```

Comment:
I prefer to always have a space after keywords and you should do the assignment on a new line.